client-grpc/src/macros.rs

```diff
@@ -1,40 +1,1103 @@
+                    grpc_warn!("(replace_linked) {} client.", self.get_name());
+                    grpc_debug!("(replace_linked) request: {:?}", request);
+                    self.get_client().await?.replace_linked(request).await
+                }
+
+                async fn unlink(
+                    &self,
+                    request: $crate::Id,
+                ) -> Result<tonic::Response<()>, tonic::Status> {
+                    grpc_warn!("(unlink) {} client.", self.get_name());
+                    grpc_debug!("(unlink) request: {:?}", request);
+                    self.get_client().await?.unlink(request).await
+                }
+
+                async fn get_linked_ids(
+                    &self,
+                    request: $crate::Id,
+                ) -> Result<tonic::Response<$crate::IdList>, tonic::Status> {
+                    grpc_warn!("(get_linked_ids) {} client.", self.get_name());
+                    grpc_debug!("(get_linked_ids) request: {:?}", request);
+                    self.get_client().await?.get_linked_ids(request).await
+                }
+
+                async fn get_linked(
+                    &self,
+                    request: $crate::Id,
+                ) -> Result<tonic::Response<Self::OtherList>, tonic::Status> {
+                    grpc_warn!("(get_linked) {} client.", self.get_name());
+                    grpc_debug!("(get_linked) request: {:?}", request);
+                    self.get_client().await?.get_linked(request).await
+                }
+
+                async fn is_ready(
+                    &self,
+                    request: $crate::ReadyRequest,
+                ) -> Result<tonic::Response<$crate::ReadyResponse>, tonic::Status> {
+                    grpc_warn!("(is_ready) {} client.", self.get_name());
+                    grpc_debug!("(is_ready) request: {:?}", request);
+                    self.get_client().await?.is_ready(request).await
+                }
+            }
+        )+
+    };
+}
+
+/// Generates Client implementation for link gRPC MOCK clients
+#[cfg(feature = "stub_client")]
+#[macro_export]
+macro_rules! link_grpc_client {
+    ($($resource:ident, $rpc_link_client:ident, $link_object:ident, $other_resource:ident),+) => {
+        $(
+            #[tonic::async_trait]
+            impl $crate::LinkClient<$rpc_link_client<Channel>> for GrpcClient<$rpc_link_client<Channel>> {
+                type LinkObject = $resource::$link_object;
+                type OtherList = $other_resource::List;
+
+                async fn link(
+                    &self,
+                    request: Self::LinkObject,
+                ) -> Result<tonic::Response<()>, tonic::Status> {
+                    grpc_warn!("(link MOCK) {} client.", self.get_name());
+                    grpc_debug!("(link MOCK) request: {:?}", request);
+                    let id = request.id;
+                    let other_ids = request.other_id_list.unwrap();
```

Comment:
Some unhandled unwraps() in the following files:
- server/src/grpc/macros.rs
- client-grpc/src/macros.rs 
- client-grpc/includes/search.rs 
- Mock files under client-grpc/includes/